ktor-http/ktor-http-cio/jvm/src/io/ktor/server/cio/backend/ServerPipeline.kt

```diff
@@ -87,7 +87,7 @@ public fun CoroutineScope.startServerConnectionPipeline(
 
             val response = ByteChannel()
 
-            val transferEncoding = request.headers["Transfer-Encoding"]
+            val transferEncoding = request.headers["Transfer-Encoding"]?.trim()
```

Comment:
`trim()` is relatively slow (because of the additional allocation and unicode checks). Can't we avoid it? Like by tuning `expectHttpBody` to do case-insensitive indexOf instead of equals?